libcxx/utils/libcxx/test/features.py

```diff
@@ -7,6 +7,7 @@
 # ===----------------------------------------------------------------------===##
 
 from libcxx.test.dsl import *
+import libcxx.test.config as config
```

Comment:
I really like how cleaner the logic is after this patch, however I would really like to avoid threading this through the CMake. Ideally, the test suite would be the one to figure out the executable to use for clang-tidy. Something like this:

```diff
diff --git a/libcxx/utils/libcxx/test/params.py b/libcxx/utils/libcxx/test/params.py
index 89d9f22e9dc6..c6b32f6fae61 100644
--- a/libcxx/utils/libcxx/test/params.py
+++ b/libcxx/utils/libcxx/test/params.py
@@ -366,6 +366,16 @@ DEFAULT_PARAMETERS = [
         default=f"{shlex.quote(sys.executable)} {shlex.quote(str(Path(__file__).resolve().parent.parent.parent / 'run.py'))}",
         help="Custom executor to use instead of the configured default.",
         actions=lambda executor: [AddSubstitution("%{executor}", executor)],
+    ),
+    Parameter(
+        name='clang-tidy-executable',
+        type=str,
+        default=_getSuitableClangTidy(),
+        help="TODO",
+        actions=lambda exe: [] if exe is None else [
+            AddSubstitution('%{clang-tidy}', exe),
+            AddFeature('has-clang-tidy')
+        ]
     )
 ]
 # fmt: on
```

Reply:
I've given this a bit more thought. One of the reasons to move it to CMake was to be able to select the proper clang library to build the plugin against. Moving this to features re-introduces that issue. Let's discuss this next time we meet.